src/eventmaster/api_handlers.go

```diff
@@ -414,37 +414,43 @@ func (h *httpHandler) handleGetDc(w http.ResponseWriter, r *http.Request, _ http
 		h.sendError(w, http.StatusInternalServerError, err, "Error marshalling response to JSON", r.URL.Path)
 		return
 	}
-	h.sendResp(w, "", string(str), r.URL.Path)
+	h.sendResp(w, str, r.URL.Path)
 }
 
-func (h *httpHandler) handleGitHubEvent(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-	var info map[string]interface{}
-
-	defer r.Body.Close()
-	reqBody, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		h.sendError(w, http.StatusBadRequest, err, "Error reading request body", r.URL.Path)
-		return
-	}
-	if err := json.Unmarshal(reqBody, &info); err != nil {
-		h.sendError(w, http.StatusBadRequest, err, "Error JSON decoding body of request", r.URL.Path)
-		return
-	}
-
-	id, err := h.store.AddEvent(&UnaddedEvent{
-		Dc:        "github",
-		Host:      "github",
-		TopicName: "github",
-		Data:      info,
-	})
-	if err != nil {
-		h.sendError(w, http.StatusInternalServerError, err, "Error adding event to store", r.URL.Path)
-		return
+func (h *httpHandler) handlePluginEvent(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	pluginName := ps.ByName("name")
+	if plugin, ok := Plugins[pluginName]; ok {
+		defer r.Body.Close()
+		rawData, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			h.sendError(w, http.StatusInternalServerError, err, "Error reading request body", r.URL.Path)
+			return
+		}
+		valid := plugin.ValidateAuth(r, rawData)
```

Comment:
if `valid` won't be used elsewhere you can combine the lines to look something like:

```
if valid:= plugin.ValidateAuth(); !valid {
    // something
}
```